### PR TITLE
Fix randomly failing ejabberd_listeners_SUITE

### PR DESCRIPTION
**Pull request by michalwski**
Proposed changes include:
* retries in waiting for listeners to be closed in ejabberd_listeners_SUITE

### CODE REVIEW

**Comment by mongoose-im** [comment]
[5064.1](https://travis-ci.org/esl/MongooseIM/jobs/400886376) / Erlang 19.3 / small_tests / 533920e4bfac93f74764f94a1148889f83b1d9e8
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/1967/5064/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1967/5064/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-07-06_13.35.00/lib.mongooseim.logs/run.2018-07-06_13.35.01/suite.log.html)

**Comment by michalwski** [comment]
> Looks good but probably should be based on changes from #1962 after it is merged.

I'm not sure. `small` and `big` tests don't share the code (unless we hack one or the other and copy modules).

File: test/async_helper.erl [hunk 1]
```diff
@@ -47,3 +47,39 @@ helper_loop() ->
...
+    Defaults = #{time_left => timer:seconds(5),
+                 sleep_time => 100,
+                 history => []},
+    do_wait_until(Fun, ExpectedValue, maps:merge(Defaults, Opts)).
+
+do_wait_until(_Fun, _ExpectedValue, #{
+                                      time_left := TimeLeft,
+                                      history := History
+                                     }) when TimeLeft =< 0 ->
+    error({badmatch, History});
```
Comment (fenek) [inline test/async_helper.erl]: History will be returned in reverse order here; might be misleading. :)

**Comment by fenek** [comment]
Ah, correct, I forgot it's a small test. So actually I've added one comment but the PR seems OK in general.

File: test/async_helper.erl (hunk 1 above)
Comment (michalwski) [reply test/async_helper.erl]: Right it might be misleading. I'll change it to return reversed list.

**Comment by mongoose-im** [comment]
[5091.1](https://travis-ci.org/esl/MongooseIM/jobs/402079216) / Erlang 19.3 / small_tests / 9bdddc34af6683ba4162e248019d64c0fee50d9f
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/1967/5091/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1967/5091/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-07-10_06.46.10/lib.mongooseim.logs/run.2018-07-10_06.46.10/suite.log.html)

**Pull request merged**